Added DoR und DoD (#6)
olr-fh/ITOPS-Studienarbeit: chapter/Cycle.tex

Diese Risiken müssen gut gehandhabt werden, wofür unser Unternehmen eine Risikomatrix verwendet, um sie nach ihrer Wahrscheinlichkeit zu bewerten, was eine frühzeitige Identifizierung ermöglicht. Um technische Risiken zu vermeiden, wird eine kontinuierliche Integration und Bereitstellung mit automatisierten Tests eingesetzt, um Fehler und Bugs zu erkennen und zu beheben. Backup-Strategien werden ebenfalls verwendet, um sicherzustellen, dass Datenverluste vermieden werden, diese sind tägliche inkrementelle Backups, die durchgeführt werden, um die letzten Änderungen zu speichern, wöchentliche vollständige Backups, um eine vollständige Wiederherstellung zu bieten, und monatliche Differentiale, um eine zusätzliche Sicherheitsebene zu bieten.

\subsubsection{Definition of Ready und Defintion of Done (Oliver)}

> Die "`Definition of Ready"' (DoR) ist ein vereinbarter Satz von Kriterien, die eine User Story erfüllen muss, bevor es in einen Sprint aufgenommen wird. Beim BlümchenHub wird sich dafür am INVEST-Prinzip orientiert. Für das Blümchenhub2.0 sieht die DoR so aus:

Reviewer: Mach die Dinger mal als Abkürzung, in LaTex wie man das richtig macht, rein.
Dinger:
 - DoR
 - INVEST
 - DoD

> Die "`Definition of Ready"' (DoR) ist ein vereinbarter Satz von Kriterien, die eine User Story erfüllen muss, bevor es in einen Sprint aufgenommen wird. Beim BlümchenHub wird sich dafür am INVEST-Prinzip orientiert. Für das Blümchenhub2.0 sieht die DoR so aus:

Reviewer: ```suggestion
Die "`Definition of Ready"' (DoR) ist ein vereinbarter Satz von Kriterien, die eine User Story erfüllen muss, bevor es in einen Sprint aufgenommen wird. Hierfür wird sich am INVEST-Prinzip orientiert.
```

> Zuerst der Geschäftswert, jede User Story sollte einen klaren Geschäftswert haben, der vom Produktbesitzer priorisiert wurde. In unserem Fall sollte es für das BlümchenHub einen Wert haben. Als nächstes sind klare Akzeptanzkriterien nötig. Die Akzeptanzkriterien für die User Story sind definiert und von allen Beteiligten verstanden. Als Nächstes sollte die User Story schätzbar sein, sie soll klein genug sein, um geschätzt und in einem Sprint abgeschlossen zu werden. Eine User Story sollte keine Abhängigkeiten haben. Die User Story hat keine externen Abhängigkeiten, die die Fertigstellung innerhalb des Sprints verhindern könnten. Außerdem sollte sie technisch durchführbar sein, das heißt das Entwicklungsteam hat die User Story überprüft und ist der Meinung, dass es technisch durchführbar ist. Zuletzt sollte die User Story testbar sein.

Reviewer: Vlt erst aufzählen, was INVEST heißt und dann sagen, was die Bestandteile sind? Das würde den roten Faden klarer machen.

Und der ganze erste Teil mit "wir müssen uns darauf einigen, dass wir das einhalten" kann glaube auch weg. Oder du trennst das textlich von der Erklärung von INVEST. Oder habe ich deine ersten Sätze komplett falisch verstanden und das gehört zur INVEST Erklärung? Ka. Finde vor allem den ersten Satz verwirrend.

> Diese Kriterien stellen sicher, dass jede User Story, das in einen Sprint aufgenommen wird, gut verstanden ist und das Entwicklungsteam zuversichtlich ist, dass es die User Story innerhalb des Sprints abschließen kann.

Reviewer: ```suggestion
Diese Kriterien stellen sicher, dass jede User Story, die in einen Sprint aufgenommen wird, gut verständlich ist und das Entwicklungsteam zuversichtlich ist, dass es die User Story innerhalb des Sprints abgeschlossen werden kann.
```

> Die "`Definition of Done"' (DoD) ist eine Liste von Kriterien, die ein Softwareprodukt oder eine Funktion erfüllen muss, bevor es als "`fertig"' betrachtet wird. So sieht die DoD beim BlümchenHub aus:

Reviewer: ```suggestion
Die "`Definition of Done"' (DoD) ist eine Liste von Kriterien, die ein Softwareprodukt oder eine Funktion erfüllen muss, bevor es als "`fertig"' betrachtet wird.
```

> Als Erstes muss der Code für das Feature vollständig geschrieben sein und die Akzeptanzkriterien erfüllen. Außerdem sollte die Code-Review abgeschlossen sein, das heißt der Code wurde von mindestens einem anderen Entwickler überprüft und genehmigt. Es sollten die nötigen Unit-Tests vorhanden sein, also wurden für den Code Unit-Tests geschrieben und alle Tests sind erfolgreich durchgelaufen. Dazu sollten Integrationstests vorhanden sein, wenn das für das hinzugefügte nötig ist. Diese sollten auch ohne Fehler durchlaufen.

Reviewer: Hier sollte noch geplänkel vor mit wir sind ein live-service-shit und deswegen niemals fertig. Folglich definieren wir die DoD nur für Features. :]

> Als Erstes muss der Code für das Feature vollständig geschrieben sein und die Akzeptanzkriterien erfüllen. Außerdem sollte die Code-Review abgeschlossen sein, das heißt der Code wurde von mindestens einem anderen Entwickler überprüft und genehmigt. Es sollten die nötigen Unit-Tests vorhanden sein, also wurden für den Code Unit-Tests geschrieben und alle Tests sind erfolgreich durchgelaufen. Dazu sollten Integrationstests vorhanden sein, wenn das für das hinzugefügte nötig ist. Diese sollten auch ohne Fehler durchlaufen.

Reviewer: ```suggestion
Als Erstes muss der Code für das Feature vollständig geschrieben sein und die Akzeptanzkriterien erfüllen. Außerdem sollte die Code-Review abgeschlossen sein, das heißt der Code wurde von mindestens einem anderen Entwickler überprüft und genehmigt. Es sollten die nötigen Unit-Tests vorhanden sein, also wurden für den Code Unit-Tests geschrieben und alle Tests sind erfolgreich durchgelaufen.
```
Is halt unnötig hier nochmal auf eine Testart einzugehen. Mit "Muss halt ein Test da sein" ist da alles gesagt.

> Diese Kriterien stellen sicher, dass das Produkt oder Feature in einem Zustand ist, der für den Einsatz bereit ist.

Reviewer: ```suggestion
Diese Kriterien stellen sicher, dass unsere entwickelten Features in einem, für die Auslieferung fertigen, Zustand sind, wenn das Work Item auf Done gestellt wurde.
```
so oder so ähnlich

> Es ist wichtig zu beachten, dass die spezifischen Kriterien für die DoR und DoD von Team zu Team variieren können, wenn das der Fall ist, können die einzelnen Teams die DoR und DoD anpassen.

Reviewer: Was ist das denn für ne geile Logik? "Wenn das varriiert, dann können die das anpassen"? Ist das nicht das selbe in anderer Schriftart?

Wie wäre es mit:
```suggestion
In manchen Fällen ist es sinnvoll die zuvor definierte DoR und DoD zu verändern, weshalb es den Teams freigestellt ist diese anzupassen. Bei solchen Anpassungen sollte jedoch kritisch hinterfragt werden, ob diese wirklich Sinn machen.
```

Diese Risiken müssen gut gehandhabt werden, wofür unser Unternehmen eine Risikomatrix verwendet, um sie nach ihrer Wahrscheinlichkeit zu bewerten, was eine frühzeitige Identifizierung ermöglicht. Um technische Risiken zu vermeiden, wird eine kontinuierliche Integration und Bereitstellung mit automatisierten Tests eingesetzt, um Fehler und Bugs zu erkennen und zu beheben. Backup-Strategien werden ebenfalls verwendet, um sicherzustellen, dass Datenverluste vermieden werden, diese sind tägliche inkrementelle Backups, die durchgeführt werden, um die letzten Änderungen zu speichern, wöchentliche vollständige Backups, um eine vollständige Wiederherstellung zu bieten, und monatliche Differentiale, um eine zusätzliche Sicherheitsebene zu bieten.

\subsubsection{\acrlong{DoR} und \acrlong{DoD} (Oliver)}

Die \gls{DoR} ist ein vereinbarter Satz von Kriterien, die eine User Story erfüllen muss, bevor es in einen Sprint aufgenommen wird. Hierfür wird sich am \acrshort{INVEST}-Prinzip orientiert. 

> Das \acrshort{INVEST}-Prinzip besteht aus 6 Komponenten: \acrlong{INVEST}. Zuerst soll die User Story keine externen Abhängigkeiten haben, die die Fertigstellung innerhalb des Sprints verhindern könnten. Dazu soll die User Story verhandelbar sein, sie soll somit dem Entwickler genug Freiheiten lassen. Außerdem muss jede User Story einen klaren Geschäftswert haben, der vom Produkt Owner priorisiert wurde. Als Nächstes sollte die User Story schätzbar sein, dafür soll sie klein genug sein, um geschätzt und in einem Sprint abgeschlossen zu werden. Zuletzt sollte die User Story testbar sein.

Reviewer: "Unabhängig" hat nicht nur mit externen Abhängigkeiten zu tun. Was auch immer extern in diesem Kontext übnerhaupt heißt. Außerhalb des Sprints?

"Zuerst soll eine User Story Unabhängig von anderen User Stroies sein. Somit wird sichergestellt, dass nicht erst auf die Fertigstellung einer anderen Story gewartet werden muss und Produktivitäts-Bottlenecks werden vermieden."

So eine Erklärung wäre glaube nen bissel genauer. Eins zu eins würde ich die aber nicht übernehmen, da Bottleneck nicht wirklich erklärt wird usw.
Auch bei den anderen Prinzipien würde ich die ein bissel besser/ genauer erklären, als die noch ein zweites mal zu nennen.

> Ergänzend zum \acrshort{INVEST}-Prinzip sollte die User Story technisch durchführbar sein, das heißt das Entwicklungsteam hat die User Story überprüft und ist der Meinung, dass es technisch durchführbar ist. Außerdem sind klare Akzeptanzkriterien nötig. Die Akzeptanzkriterien für die User Story sind definiert und von allen Beteiligten verstanden. 

Reviewer: ```suggestion
Ergänzend zum \acrshort{INVEST}-Prinzip sollte die User Story technisch durchführbar sein, das heißt, dass das Entwicklungsteam die User Story überprüft hat und der Meinung ist, dass diese technisch durchführbar ist. Außerdem sind klare Akzeptanzkriterien nötig. Die Akzeptanzkriterien für die User Story sind definiert und von allen Beteiligten verstanden. 
```
Letzter Satz, eine DoD ist da?

Diese Kriterien stellen sicher, dass jede User Story, die in einen Sprint aufgenommen wird, gut verständlich ist und das Entwicklungsteam zuversichtlich ist, dass die User Story innerhalb des Sprints abgeschlossen werden kann.

> Die \gls{DoD} ist eine Liste von Kriterien, die ein Softwareprodukt oder eine Funktion erfüllen muss, bevor es als "`fertig"' betrachtet wird. Die BlümchenHub2.0 Website ist ein live-service Modell, daher ist es als Produkt niemals "`fertig"', sondern wird immer erweitert. Daher gilt die \gls{DoD} für das BlümchenHub2.0 nur für Features.

Reviewer: live-service ist was von Spielen und man weiß auch vlt nicht so ganz was damit geneint ist

> Als Erstes muss der Code für das Feature vollständig geschrieben sein und die Akzeptanzkriterien erfüllen. Außerdem sollte die Code-Review abgeschlossen sein, das heißt der Code wurde von mindestens einem anderen Entwickler überprüft und genehmigt. Es sollten alle nötigen Tests vorhanden sein. Diese müssen ohne Fehler durchlaufen.

Reviewer: welche "Aktzeptanzkriterien"?
